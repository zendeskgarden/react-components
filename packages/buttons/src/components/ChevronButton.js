/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React from 'react';
import PropTypes from 'prop-types';
import IconButton from './IconButton';
import ChevronDownIcon from '@zendeskgarden/svg-icons/src/16/chevron-down-stroke.svg';

const SIZE = {
  SMALL: 'small',
  MEDIUM: 'medium',
  LARGE: 'large'
};

/**
 * An `IconButton` with an embedded chevron icon
 */
const ChevronButton = React.forwardRef(({ ...buttonProps }, ref) => (
  <IconButton ref={ref} {...buttonProps}>
    <ChevronDownIcon />
  </IconButton>
));

ChevronButton.propTypes = {
  /** Apply danger styling */
  danger: PropTypes.bool,
  size: PropTypes.oneOf([SIZE.SMALL, SIZE.MEDIUM, SIZE.LARGE]),
  /** Applies primary button styling */
  primary: PropTypes.bool,
  /** Applies basic button styling */
  basic: PropTypes.bool,
  /** Applies pill styling */
  pill: PropTypes.bool,
  /** Applies inset `box-shadow` styling on focus */
  focusInset: PropTypes.bool,
  /** Rotates icon 180 degrees */
  rotated: PropTypes.bool
};

ChevronButton.defaultProps = {
  basic: false,
  pill: false,
  size: SIZE.MEDIUM
};

export default ChevronButton;
