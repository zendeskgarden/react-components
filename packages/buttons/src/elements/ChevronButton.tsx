/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React from 'react';
import IconButton, { IIconButtonProps } from './IconButton';
import ChevronDownIcon from '@zendeskgarden/svg-icons/src/16/chevron-down-stroke.svg';

/**
 * @extends ButtonHTMLAttributes<HTMLButtonElement>
 */
const ChevronButton: React.FunctionComponent<
  IIconButtonProps & React.RefAttributes<HTMLButtonElement>
> = React.forwardRef<HTMLButtonElement, IIconButtonProps>(({ ...buttonProps }, ref) => (
  <IconButton ref={ref} {...buttonProps}>
    <ChevronDownIcon />
  </IconButton>
));

ChevronButton.displayName = 'ChevronButton';

ChevronButton.propTypes = IconButton.propTypes;

ChevronButton.defaultProps = {
  isBasic: false,
  isPill: false,
  size: 'medium'
};

export default ChevronButton;
