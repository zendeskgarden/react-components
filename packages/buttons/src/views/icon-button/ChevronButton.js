/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React from 'react';
import PropTypes from 'prop-types';
import IconButton from './IconButton';
import Icon from './Icon';
import ChevronDownIcon from '@zendeskgarden/svg-icons/src/16/chevron-down-stroke.svg';

const SIZE = {
  SMALL: 'small',
  LARGE: 'large'
};

/**
 * IconButton with an embedded chevron icon
 */
const ChevronButton = ({ rotated, ...buttonProps }) => (
  <IconButton pill={false} muted={false} basic={false} {...buttonProps}>
    <Icon rotated={rotated}>
      <ChevronDownIcon />
    </Icon>
  </IconButton>
);

ChevronButton.propTypes = {
  /** Apply danger styling */
  danger: PropTypes.bool,
  size: PropTypes.oneOf([SIZE.SMALL, SIZE.LARGE]),
  /** Applies primary button styling */
  primary: PropTypes.bool,
  /** Applies basic button styling */
  basic: PropTypes.bool,
  /** Applies muted button styling */
  muted: PropTypes.bool,
  /** Applies pill styling */
  pill: PropTypes.bool,
  disabled: PropTypes.bool,
  focused: PropTypes.bool,
  /** Applies inset `box-shadow` styling on focus */
  focusInset: PropTypes.bool,
  hovered: PropTypes.bool,
  active: PropTypes.bool,
  /** Callback for reference of the native button element */
  buttonRef: PropTypes.func,
  /** Rotates icon 180 degrees */
  rotated: PropTypes.bool
};

export default ChevronButton;
