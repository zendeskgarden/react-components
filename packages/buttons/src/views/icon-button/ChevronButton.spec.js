/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React from 'react';
import { shallow } from 'enzyme';
import ChevronButton from './ChevronButton';

describe('ChevronButton', () => {
  it('renders IconButton with correct default styling', () => {
    const wrapper = shallow(<ChevronButton />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders focus-inset styling if provided', () => {
    const wrapper = shallow(<ChevronButton focusInset />);

    expect(wrapper).toMatchSnapshot();
  });

  it('rotates icon if prop is provided', () => {
    const wrapper = shallow(<ChevronButton rotated />);

    expect(wrapper).toMatchSnapshot();
  });
});
