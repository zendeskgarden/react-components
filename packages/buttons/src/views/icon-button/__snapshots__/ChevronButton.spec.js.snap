// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChevronButton renders IconButton with correct default styling 1`] = `
<IconButton
  basic={false}
  muted={false}
  pill={false}
>
  <Icon>
    <MockSVG />
  </Icon>
</IconButton>
`;

exports[`ChevronButton renders focus-inset styling if provided 1`] = `
<IconButton
  basic={false}
  focusInset={true}
  muted={false}
  pill={false}
>
  <Icon>
    <MockSVG />
  </Icon>
</IconButton>
`;

exports[`ChevronButton rotates icon if prop is provided 1`] = `
<IconButton
  basic={false}
  muted={false}
  pill={false}
>
  <Icon
    rotated={true}
  >
    <MockSVG />
  </Icon>
</IconButton>
`;
